addons/point_of_sale/controllers/main.py

```diff
@@ -33,8 +33,10 @@ def pos_web(self, config_id=False, **k):
                 ('user_id', '=', request.session.uid),
                 ('rescue', '=', False)
                 ]
-        if config_id:
+        if config_id and config_id.isnumeric():
             domain = AND([domain,[('config_id', '=', int(config_id))]])
+        else:
+            return request.not_found()
```

Comment:
I think we don't need to return an error if config_id is not set, the code seems to support this case

```suggestion
```

But fixing an error that happens because of a badly formatted url I don't think is worth it. We can cause a traceback everywhere like that, we can't fix everything. A more global management of types in urls is needed